lib/puppet/info_service/task_information_service.rb

```diff
@@ -4,11 +4,17 @@ class Puppet::InfoService::TaskInformationService
   def self.tasks_per_environment(environment_name)
     # get the actual environment object, raise error if the named env doesn't exist
     env = Puppet.lookup(:environments).get!(environment_name)
+
     env.modules.map do |mod|
       mod.tasks.map do |task|
-        {:module => {:name => task.module.name}, :name => task.name, :metadata => task.metadata}
+        # If any task is malformed continue to list other tasks in module
+        begin
+          task.validate
+          {:module => {:name => task.module.name}, :name => task.name, :metadata => task.metadata}
+        rescue Puppet::Module::Task::Error => _
+        end
```

Comment:
Can we log a warning with the error and have an implicit `nil` at the end so it's clear it's compacted later.